rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/BaseWorkspaceService.java

```diff
@@ -112,6 +135,17 @@ public void didChangeWorkspaceFolders(DidChangeWorkspaceFoldersParams params) {
         }
     }
 
+    @Override
+    public void didRenameFiles(RenameFilesParams params) {
+        logger.debug("workspace/didRenameFiles: {}", params.getFiles());
+
+        CompletableFuture.supplyAsync(() -> workspaceFolders()
+                .stream()
+                .map(f -> Locations.toLoc(f.getUri()))
+                .collect(Collectors.toSet()), getExecuter())
+                .thenAccept(folders -> documentService.didRenameFiles(params, folders));
```

Comment:
nit: the translation of `workspaceFolders` to a `Set<ISourceLocation>` can also be done in the same block as the `documentService.didRenameFiles`. And I would even say that just passing in the result of `workspaceFolders` to the `didRenameFiles` function would be my prefered solution, so we don't do conversion when we don't even need it (and it's all just lsp4j classes anyway).

Reply:
I agree, I have changed the implementation to use `List<WorkspaceFolder>` instead.